pkg/interactive/docker.go

```diff
@@ -76,39 +74,36 @@ func (d *Docker) AddMount(typeStr, source, target string, optAdditionalArgs ...s
 	d.AddArgs("--mount", fmt.Sprintf("type=%s,source=%s,target=%s%s", typeStr, source, target, addl))
 }
 
-// AddRWOverlay mounts a directory into the image at the desired location, but with an overlay
-//  so internal changes do not modify the external directory.
-// externalDirectory probably needs to be an absolute path
-// Returns a function to clean up the mount (but does not delete the directory).
-// Uses sudo and probably only works on Linux
-func (d *Docker) AddRWOverlay(externalDirectory, internalDirectory string) func() {
-	tmpDir, err := ioutil.TempDir("", "overlay")
+// CopyAndAddMount copies the source files into a temp directory, and then
+// mounts them as the target. It also returns a function to remove the temp
+// directory for cleaning up.
+func (d *Docker) CopyAndAddMount(typeStr, parentDir, source, target string, optAdditionalArgs ...string) (func(), error) {
+	fi, err := os.Stat(source)
 	if err != nil {
-		log.Fatal(err)
+		return nil, fmt.Errorf("error getting the FileInfo for %q: %w", source, err)
```

Comment:
The purpose of this library is to write interactive scripts, not programs which can recover from errors and do smart things. Adding error checking makes the scripts much less focused on the task at hand and hard to parse.

If you want different functionality, put it in a different library

Reply:
Hmm I'm not quite sure what makes the difference between them, and we already have another function that returns error instead of fatal - https://github.com/knative/test-infra/blob/master/pkg/interactive/docker.go#L43-L56

Reply:
Because that error is optional -- sometimes it's okay to fail and sometimes it is not.

If the overlay fails, it's unlikely you want to continue

Reply:
I have change it to fatal instead of returning an error.